crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -139,7 +139,13 @@ impl SettlementEncoder {
             tokens: self.tokens.clone(),
             clearing_prices: self.clearing_prices.clone(),
             trades: self.trades.clone(),
-            execution_plan: Vec::new(),
+            execution_plan: self
+                .execution_plan
+                .iter()
+                // Instead of simply dropping the executions we mark all the interactions as
+                // internalizable.
+                .map(|(execution, _)| (execution.clone(), true))
+                .collect(),
             pre_interactions: self.pre_interactions.clone(),
```

Comment:
What is actually the difference here if you mark all the interactions internalizable and then get them dropped in the `finish`?

Reply:
The difference is where the interactions gets dropped. If we only mark the interaction as internalizable we will simulate the original and optimized solution during the settlement ranking which will give us better insight in the actual solution. But if the interaction already got dropped there, we could end up with a solution where the original version would revert but the optimized one would succeed erroneously (see example in the description).

Another reason to use the flag instead of dropping the interaction is that this is how external solvers are supposed to do it exactly because we simulate to determine whether those interactions are even allowed to be internalized.

Reply:
Ah yes, I forgot you wrapped all the solvers in `OptimizingSolver`. In my head it was still the idea optimization happening just at the end of the auction process before submission.